### PR TITLE
Add IstioOperator external API

### PR DESCRIPTION
**Pull request by jahvon**
Adds the IstioOperator API to the external istio APIs we are importing. (Needed for generating code with this API)

Depends on https://github.com/solo-io/skv2/pull/293

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -6,22 +6,22 @@ replace (
...
 	github.com/servicemeshinterface/smi-sdk-go v0.4.1
 	github.com/solo-io/skv2 v0.18.3
-	istio.io/client-go v0.0.0-20200610222318-1cfead1f1938
-	k8s.io/api v0.19.7
-	k8s.io/apiextensions-apiserver v0.19.7
-	k8s.io/apimachinery v0.19.7
-	k8s.io/client-go v0.19.7
-	sigs.k8s.io/controller-runtime v0.7.0
+	istio.io/client-go v0.0.0-20210713195242-59adb375bae9
+	istio.io/istio v0.0.0-20210719174534-49dfd655557f
```
Comment (jahvon) [inline go.mod:21]: Pulling in this dependency bumped the version of a lot of the other dependencies - which resulted in a larger codegen diff

File: .github/workflows/pull_request.yaml [hunk 2]
```diff
@@ -11,10 +11,10 @@ jobs:
     name: Generate Code
     runs-on: ubuntu-18.04
     steps:
-    - name: Set up Go 1.14
+    - name: Set up Go 1.16
```
Comment (jahvon) [inline .github/workflows/pull_request.yaml:14]: Given the newer dependencies, codegen was failing for Golang version 1.14 with errors like
`embed: package embed is not in GOROOT`
`testing/fstest: package testing/fstest is not in GOROOT`
`io/fs: package io/fs is not in GOROOT`

File: go.mod [hunk 3]
```diff
@@ -1,27 +1,28 @@
...
 	github.com/servicemeshinterface/smi-sdk-go v0.4.1
 	github.com/solo-io/skv2 v0.18.3
-	istio.io/client-go v0.0.0-20200610222318-1cfead1f1938
-	k8s.io/api v0.19.7
-	k8s.io/apiextensions-apiserver v0.19.7
-	k8s.io/apimachinery v0.19.7
-	k8s.io/client-go v0.19.7
-	sigs.k8s.io/controller-runtime v0.7.0
+	istio.io/client-go v0.0.0-20210713195242-59adb375bae9
+	istio.io/istio v0.0.0-20210720120633-f8c8dba44cb7
```
Comment (jahvon) [inline go.mod:22]: Pulling in this dependency bumped the version of a lot of the other dependencies - which resulted in a larger codegen diff

File: pkg/api/istio/install.istio.io/v1alpha1/clients.go [hunk 4]
```diff
@@ -0,0 +1,214 @@
...
+	IstioOperators() IstioOperatorClient
+}
+
+type clientSet struct {
+	client client.Client
+}
+
+func NewClientsetFromConfig(cfg *rest.Config) (Clientset, error) {
+	scheme := scheme.Scheme
+	if err := install_istio_io_v1alpha1.AddToScheme(scheme); err != nil {
```
Comment (jahvon) [inline pkg/api/istio/install.istio.io/v1alpha1/clients.go:53]: `install_istio_io_v1alpha1` does not define `AddToScheme`

File: codegen/istio_groups.go [hunk 5]
```diff
@@ -4,6 +4,7 @@ import (
 	"github.com/solo-io/skv2/codegen/model"
 	"istio.io/client-go/pkg/apis/networking/v1alpha3"
 	"istio.io/client-go/pkg/apis/security/v1beta1"
+	operator "istio.io/istio/operator/pkg/apis/istio/v1alpha1"
```
Comment (jahvon) [inline codegen/istio_groups.go:7]: IstioOperator API is not in istio/client-go so I have to import it from istio/istio

File: .github/workflows/pull_request.yaml (hunk 2 above)
Comment (jahvon) [inline .github/workflows/pull_request.yaml:14]: A dependency of the newer version of skv2 had a dependency that didn't work with Go 1.14. I was getting the error `io/fs: package io/fs is not in GOROOT (/opt/hostedtoolcache/go/1.14.15/x64/src/io/fs)` in CI

File: go.mod [hunk 6]
```diff
@@ -1,27 +1,29 @@
...
 	github.com/pkg/errors v0.9.1
 	github.com/rotisserie/eris v0.1.1
 	github.com/servicemeshinterface/smi-sdk-go v0.4.1
-	github.com/solo-io/skv2 v0.18.3
-	istio.io/client-go v0.0.0-20200610222318-1cfead1f1938
-	k8s.io/api v0.19.7
-	k8s.io/apiextensions-apiserver v0.19.7
-	k8s.io/apimachinery v0.19.7
-	k8s.io/client-go v0.19.7
+	github.com/solo-io/skv2 v0.19.8-0.20210908202014-78e62d569a05
```
Comment (jahvon) [inline go.mod:21]: Can not merge until skv2 change is merged and released

File: go.mod [hunk 7]
```diff
@@ -1,27 +1,29 @@
...
 
 replace (
 	// pinned to solo-io's fork of cue version 95a50cebaffb4bdba8c544601d8fb867990ad1ad
 	cuelang.org/go => github.com/solo-io/cue v0.4.1-0.20210622213027-95a50cebaffb
 
-	github.com/Azure/go-autorest => github.com/Azure/go-autorest v13.0.0+incompatible
 	github.com/Sirupsen/logrus => github.com/sirupsen/logrus v1.4.2
 	github.com/docker/docker => github.com/moby/moby v0.7.3-0.20190826074503-38ab9da00309
+
+	github.com/golang/mock => github.com/golang/mock v1.4.4
```
[the remark this answers is not in the document]
Comment (EItanya) [reply go.mod:12]: I think we're good to bump, we use 1.6.x elsewhere!

[the remark this answers is not in the document]
Comment (jahvon) [reply go.mod]: Removed this replace!

**Review by EItanya — APPROVED** [review]

**Pull request merged**